client/src/search.rs

```diff
@@ -71,4 +73,13 @@ impl LdapClient {
             };
         }
     }
+
+    pub async fn search<S: Into<String>>(
+        &mut self,
+        basedn: S,
```

Comment:
You could use the same `impl Into<String>` here, we don't otherwise use S

Reply:
Yes we could do that but that would technically be a breaking change. And I don't now what the policy around that for this project is.

Reply:
Would it really be a breaking change? What code would stop compiling? AFAIU this is equivalent syntactic sugar

Reply:
Code that uses that explicitly uses that generic parameter like `search::<String>` would stop to compile

Reply:
Fair enough. That's a bit of an edge case, but it is indeed a breaking change